src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -82,12 +82,25 @@ internal RabbitMQContext CreateContext(RabbitMQAttribute attribute)
                 Properties = properties,
             };
 
+            IModel channel = CreateConnectionAndChannel(hostname);
+            QueueDeclareOk queue = channel.QueueDeclare(queue: queuename, durable: false, exclusive: false, autoDelete: false, arguments: null);
+            IBasicPublishBatch messages = channel.CreateBasicPublishBatch();
+
             return new RabbitMQContext

```

Comment:
Also not sure if it's better to be passing the channel directly into the context and creating the batch and message queue inside AsyncCollector or creating them here so that they are just used inside AsyncCollector